extensions/ql-vscode/src/quick-query.ts

```diff
@@ -109,24 +106,22 @@ export async function displayQuickQuery(
       libraryPathDependencies: [qlpack]
     };
 
+    // Only recreate the query file if it doesn't already exist.
+    // Always recreate the qlpack file because the file will change when database languages change
     const qlFile = path.join(queriesDir, QUICK_QUERY_QUERY_NAME);
+    if (!await fs.pathExists(qlFile)) {
+      await fs.writeFile(qlFile, getInitialQueryContents(dbItem.language, dbscheme), 'utf8');
+    }
+
     const qlPackFile = path.join(queriesDir, 'qlpack.yml');
-    await fs.writeFile(qlFile, getInitialQueryContents(dbItem.language, dbscheme), 'utf8');
-    await fs.writeFile(qlPackFile, yaml.safeDump(quickQueryQlpackYaml), 'utf8');
+    await fs.writeFile(qlPackFile, QLPACK_FILE_HEADER + yaml.safeDump(quickQueryQlpackYaml), 'utf8');
     Window.showTextDocument(await workspace.openTextDocument(qlFile));
-  }
 
-  // TODO: clean up error handling for top-level commands like this
-  catch (e) {
-    if (e instanceof UserCancellationException) {
-      logger.log(e.message);
-    }
-    else if (e instanceof ResponseError && e.code == ErrorCodes.RequestCancelled) {
-      logger.log(e.message);
-    }
-    else if (e instanceof Error)
-      showAndLogErrorMessage(e.message);
-    else
+  } catch (e) {
+    if (e instanceof ResponseError && e.code == ErrorCodes.RequestCancelled) {
+      throw new UserCancellationException(e.message);
+    } else {
       throw e;
```

Comment:
This will go to the top-level handler that logs it before displaying, right?

Reply:
That's right. This is aligning the command with all the other commands.

I should probably also take `if (e instanceof ResponseError && e.code == ErrorCodes.RequestCancelled)` and put it into the global handler.